core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/clazz/detailoverview/ClazzDetailOverviewViewModel.kt

```diff
@@ -110,6 +112,7 @@ class ClazzDetailOverviewViewModel(
     )
 
     val listRefreshCommandFlow: Flow<RefreshCommand> = _listRefreshCommandFlow.asSharedFlow()
+    private val canAddNewCourse: Boolean = savedStateHandle[UstadView.ARG_CAN_ADD_COURSE]?.toBoolean() ?: false
```

Comment:
As mentioned: this is the wrong approach. You need to collect the flow / make the permission check within ClazzDetailOverviewViewModel (as is done by all other viewmodels). You must not rely on an argument coming in from other view models for this.